tests/test_result.py

```diff
@@ -213,6 +213,26 @@ def test_and_then() -> None:
     assert Err(3).and_then(sq_lambda).and_then(sq_lambda).err() == 3
 
 
+def test_inspect() -> None:
+    oks: List[int] = []
+    assert Ok(2).inspect(lambda x: oks.append(x)) == Ok(2)
```

Comment:
Can you define these lambda's similar to the `sq_lambda` that is defined at the bottom of this file. Just because it's not possible to annotate the `x` here nicely and my editor gives me poor type checking here currently

![image](https://github.com/rustedpy/result/assets/15352422/103a4e5e-72b6-4144-8680-757c2e1d6867)

Reply:
It can't be moved to the bottom of the file since every fn is a closure there. We can't test a pure function, because a pure function in `inspect*` methods is no-op.
But I type-hinted them

Reply:
You're right. My mistake